python/pyarrow/_dataset.pyx

```diff
@@ -1075,10 +1075,11 @@ cdef class FileSystemDataset(Dataset):
             self.partition_expression
         )
 
-    @classmethod
-    def from_paths(cls, paths, schema=None, format=None,
-                   filesystem=None, partitions=None, root_partition=None):
-        """A Dataset created from a list of paths on a particular filesystem.
+    @staticmethod
+    def from_paths(paths, schema=None, format=None, filesystem=None,
+                   partitions=None, root_partition=None):
```

Comment:
Hmm... classmethod doesn't work correctly on Cython 3 anymore? Is it a known issue? Or is this change actually not needed?

Reply:
classmethod does work, I can change it back. I tried switching to staticmethod when numpydocs were failing, but it turns out it was the comment that was causing numpydoc parsing errors. I thought staticmethod was a slight improvement since `cls` wasn't actually used in the classmethod.